analyzers/tests/SonarAnalyzer.UnitTest/Common/RuleTest.cs

```diff
@@ -122,129 +107,147 @@ public void SonarDiagnosticAnalyzer_IsUsedInAllRules()
                 }
             }
         }
 
         [TestMethod]
-        public void AllCSharpRules_HaveCSharpTag()
-        {
-            GetSupportedDiagnostics(AnalyzerLanguage.CSharp)
-                .ToList()
-                .ForEach(diagnostic => diagnostic.CustomTags.Should().Contain(LanguageNames.CSharp));
-        }
+        public void AllCSharpRules_HaveCSharpTag() =>
+            SupportedDiagnostics(AnalyzerLanguage.CSharp).Should().OnlyContain(diagnostic => diagnostic.CustomTags.Contains(LanguageNames.CSharp));
 
         [TestMethod]
-        public void AllVbNetRules_HaveVbNetTag()
-        {
-            GetSupportedDiagnostics(AnalyzerLanguage.VisualBasic)
-                .ToList()
-                .ForEach(diagnostic => diagnostic.CustomTags.Should().Contain(LanguageNames.VisualBasic));
-        }
+        public void AllVbNetRules_HaveVbNetTag() =>
+            SupportedDiagnostics(AnalyzerLanguage.VisualBasic).Should().OnlyContain(diagnostic => diagnostic.CustomTags.Contains(LanguageNames.VisualBasic));
 
         [TestMethod]
         public void AllRules_SonarWayTagPresenceMatchesIsEnabledByDefault()
         {
-            var allAnalyzers =new RuleFinder().AllAnalyzerTypes.SelectMany(type => GetSupportedDiagnostics(type));
-
-            var parameterizedAnalyzers = new RuleFinder().AllAnalyzerTypes
+            var parameterized = new RuleFinder().AllAnalyzerTypes
                 .Where(RuleFinder.IsParameterized)
-                .Select(type => (DiagnosticAnalyzer)Activator.CreateInstance(type))
-                .SelectMany(analyzer => analyzer.SupportedDiagnostics)
+                .SelectMany(type => ((DiagnosticAnalyzer)Activator.CreateInstance(type)).SupportedDiagnostics)
                 .ToHashSet();
 
-            foreach (var analyzer in allAnalyzers)
+            foreach (var diagnostic in new RuleFinder().AllAnalyzerTypes.SelectMany(SupportedDiagnostics))
             {
-                var isInSonarWay = analyzer.CustomTags.Contains(DiagnosticDescriptorBuilder.SonarWayTag);
-
-                if (IsSecurityHotspot(analyzer))
+                if (IsSecurityHotspot(diagnostic))
                 {
                     // Security hotspots are enabled by default, but they will report issues only
                     // when their ID is contained in SonarLint.xml
-                    analyzer.IsEnabledByDefault.Should().BeTrue($"{analyzer.Id} should be enabled by default");
+                    diagnostic.IsEnabledByDefault.Should().BeTrue($"{diagnostic.Id} should be enabled by default");
+                }
+                else if (IsDeprecated(diagnostic))
+                {
+                    // Deprecated rules should be removed from SonarWay
+                    diagnostic.IsEnabledByDefault.Should().BeFalse($"{diagnostic.Id} is deprecated and should be disabled by default (removed from SonarWay)");
```

Comment:
how do we handle deprecated hotspots? it looks to me that we will not reach this check for them

Reply:
This check goes second, because deprecated hotspots are also active by default. That's not a problem as it's SonarLint.xml responsibility to decide if they should run or not (and they will not run in SL nor NuGet)

Reply:
Isn't the case with a deprecated hotspot in `Sonar Way` something we want to avoid?

Reply:
That's a good point, but I'm it needs some more testing and investigation => will do later.

Reply:
I don't understand why. Isn't this PR supposed to do this? 

> Write a test that will do a cross check for deprecated rules - they should not be part of SonarWay (activated by default)

Reply:
I've investigate the problem, added UT for this scenario and few comments. There are different states
`SonarWay` - is in SonarWay or not according to Jira => has SonarWay custom tag
`IsActivatedByDefault` - resx key, is true for SonarWay rules
`IsEnabledByDefault` - diagnostic state, is true for nonparametrized SonarWay rules OR all hotspots (deprecated or not)